### PR TITLE
[feature] Add Device Metadata

### PR DESCRIPTION
**Pull request by kennyklaviyo**
# Description

* Create new DeviceProperties object
* Move User Agent from KlaviyoConfig to DeviceProperties
# Check List

- [ ] Are you changing anything with the public API?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you tested this change on real device?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all Android versions the SDK currently supports?

### CODE REVIEW

**Review by evan-masseau — APPROVED** [review]
Approved but lets wait on even merging this till serverside handling of metadata in events is also ready.

File: sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    private val packageInfo: PackageInfo by lazy {
+        Registry.config.applicationContext.packageManager.getPackageInfoCompat(applicationId)
+    }
+
+    fun buildMetaData(): Map<String, String?> = mapOf(
+        "Device Manufacturer" to manufacturer,
+        "Device Model" to model,
+        "OS Name" to platform,
+        "OS Version" to osVersion,
+        "SDK Version" to sdkVersion,
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt:62]: SDK Name is missing?

File: sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        "App Version" to appVersion,
+        "App Build" to appVersionCode,
+        "Application ID" to applicationId,
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt:65]: And tiny nit, but we should pick `App` or `Application` for all 4 here.

Comment (evan-masseau) [reply sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt]: Pushed a little commit to fix these

**Review by evan-masseau — APPROVED** [review]
oh i think i did, i just pushed another commit and it re-requested approval

**Comment by evan-masseau** [comment]
TODO just noticed we're missing device identifier from the spec, will push another update today

File: sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    val osVersion: String by lazy {
+        Build.VERSION.SDK_INT.toString()
+    }
+
+    val appVersion: String by lazy {
+        packageInfo.versionName
+    }
+
+    val appVersionCode: String by lazy {
```
Comment (ajaysubra) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/DeviceProperties.kt:31]: many of these are obvious but some like this one may not be. Would it make sense to add some inline comments?

**Review by ajaysubra — APPROVED** [review]

**Comment by evan-masseau** [comment]
closing to consolidate PRs

**Pull request closed without merging**